src/Parlot/Fluent/Separated.cs

```diff
@@ -5,18 +5,20 @@
 
 namespace Parlot.Fluent
 {
-    public sealed class Separated<U, T> : Parser<List<T>>, ICompilable
+    public sealed class Separated<U, T, TParseContext, TChar> : Parser<List<T>, TParseContext, TChar>, ICompilable<TParseContext, TChar>
+    where TParseContext : ParseContextWithScanner<Scanner<TChar>, TChar>
```

Comment:
`ParseContextWithScanner` should only be necessary in low level parsers that have to deal with chars (Literals/Terms).

Reply:
I thouht about it too, but it needs to have a scanner to reset position

Reply:
and getting rid of the type makes it hard to apply Then/And/Or/... with "really scanner dependent" parsers